R/mz_summary.R

```diff
@@ -0,0 +1,171 @@
+#' @name mz_summary
+#'
+#' @title Create a summary from adjacency list containing mass-differences
+#'
+#' @description
+#' The function `mz_summary` creates a summary from the `AdjacencyMatrix`, 
+#' containing mass-differences.Individual mass differences are count over 
+#' all features. The input may be  `AdjacencyMatrix` originating from the 
+#' function `structural`, or `combine`. The parameter `filtered` will define 
+#' if data will be filtered above a certain threshold or not. 
+#' 
+#' @param am
+#' `AdjacencyMatrix`, a formal class of `AdjacencyMatrix` containing the 
+#' mass-differences, that have previously been generated by the function 
+#' `structural`or `combine`
+#' 
+#' @param filter
+#' `numeric`, leave empty or set to `0` if unfiltered data are 
+#' required. Select a `numeric` as a threshold on counts of mz differences.
+#' May be useful to visualize big data. 
+#' 
+#' @details
+#' Summarizes the adjacency matrices containing mass difference values, 
+#' i.e. either adjacency list from `structural`or `combine` may be used.
+#' The default is filter = F, so the unfiltered summary will be returned. 
+#' If filter is set to a `number`, e.g. 1000 only mz differences above 
+#' this threshold will be displayed. 
+#' The function can be applied for adjacency lists from `structural` and `combine`
+#' 
+#' @return 
+#' `data.frame` containing the numbers of present mz differences and
+#' corresponding name. 
+#'
+#' @author Liesa Salzer, \email{liesa.salzer@@helmholtz-muenchen.de}
+#'
+#' @examples
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' # unfiltered mz difference counts
+#' mz_summary(am_struct)
+#' # filtered mz difference counts 
+#' mz_summary(am_struct, filter = 2)
+#'
+#'@export
+#'
+#'@importFrom dplyr `%>%`
+mz_summary <- function(am, filter = 0){
+  
+  if (!is(am, "AdjacencyMatrix")) {
+    stop("'am' is not an 'AdjacencyMatrix' object")
+  }
+  
+  if (!validObject(am)) 
+    stop("'am' must be a valid 'AdjacencyMatrix' object") 
+  
+  am_df <- as.data.frame(am)
+  
+  if (!"combine" %in% type(am) & !"structural" %in% type(am) ) 
+    stop("'am' is not a type 'structural' or 'combine'")
+  
+  if (!1 %in% am_df$binary) 
```

Comment:
> I am not sure if I got your comment right.. So I was thinking of a function that checks if combine or structural contain any mass differences (that means any !1 %in% x_df$binary) at all. If not you will get this error message. Does it make sense?

Yes, I like this in general to test for that. But if the user modifies the object (as I did in the example, then the function will stop even though there are links in `am_df$combine_binary`. I would suggest to test separately for `am_df$binary` and `am_df$combine_binary` within each `if`/`else` statement

    if ("combine" %in% type(am)) {
        if (!1 %in% am_df$binary) 
            stop(...)
        ....
    } 
   
    else {
         if (!1 %in% am_df$combine_binary)
             stop(...)
        ....
    }

Reply:
Now I see. But for type "combine" I am using only the mass differences from am_df$combine_binary. Otherwise it would be the same as from type "structural" right?

Reply:
Therefore I would like to test: 
```

  if ("combine" %in% type(am)) {
    if (!1 %in% am_df$combine_binary) 
      stop("'am' does not contain any mass-differences")
  } 
  
  else {
    if (!1 %in% am_df$binary)
      stop("'am' does not contain any mass-differences")
  }
```

Reply:
Oh yes, I have flipped the if statements. Yours is correct - sorry